src/nodes/NodeConnectionManager.ts

```diff
@@ -426,24 +419,26 @@ class NodeConnectionManager {
    * port).
    * @param targetNodeId ID of the node attempting to be found (i.e. attempting
    * to find its IP address and port)
-   * @param timer Connection timeout timer
-   * @param options
+   * @param ignoreRecentOffline skips nodes that are within their backoff period
+   * @param ctx
    * @returns whether the target node was located in the process
    */
+  public getClosestGlobalNodes(
```

Comment:
This function is not doing the operation concurrently. This never got done concurently.

Reply:
Is this something we want to fix in this PR?

Reply:
I think I should make a new issue for this. This PR is getting too large and we should merge it sooner than later.

Reply:
Depends on how easy it is to do it. And if it can use the same pattern we are using for refresh buckets, and the semaphore... etc.

Reply:
New issue created at #448